### ISSUE TITLE
[Meta, Build] Provide working VSCode launch.json

### ISSUE THREAD

**Issue by paaspaas00** [LadybirdBrowser/ladybird#1511]
Hi all, I'm trying to debug ladybird on linux. Following the VSCode instructions, I've managed to get to a working environment where I can build in Release and in Debug mode, all working fine. However, I've got issues starting ladybird with the debugger. 
Starting debug from the CMake target (bottom CMake bar, bug icon) does not work (it should create a `launch.json` on the fly as I understand it correctly), it does not start the debugger and this is shown in the terminal:
```
error: unknown option: --interpreter=mi
error: unknown option: --tty=/dev/pts/13
Use 'lldb-17 --help' for a complete list of options.
[1] + Done(1)                    "/usr/bin/lldb-17" --interpreter=mi --tty=${DbgTerm} 0<"/tmp/Microsoft-MIEngine-In-joivakjq.kx1" 1>"/tmp/Microsoft-MIEngine-Out-jtqokwlj.dui"
```

So I tried crafting a `launch.json` manually.
Here's my attempt:
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "C++ Launch",
      "type": "cppdbg",
      "request": "launch",
      "program": "/bin/bash",
      "args": [
        "${workspaceFolder}/Meta/ladybird.sh",
        "run",
        "ladybird"
      ],
      "environment": [
        {
          "name": "config",
          "value": "Debug"
        }
      ],
      "cwd": "${workspaceFolder}"
    }
  ]
}
```
When started with `F5`, this `launch.json` is correctly picked up, the debugger is started and I can see from the Debugger pane that it sets breakpoints etc, but it doesn't stop at breakpoints, as if it's not attached to the ladybird executable or started with it.

Could you please provide a working `launch.json` for linux to compare with (or step-by-step instructions) to solve the issue?
Thank you!

**Comment by sideshowbarker** [comment]
Yeah, for debugging problems with HTML/CSS/JavaScript and web APIs (as opposed to problems with, say, the browser UI, or networking problems), you need to attach to the right WebContent process. And I'm not sure there any way to configure your environment to make gdb do that. 

But if you're able to use lldb instead, lldb has a `--waitfor` flag you can use. I know that works, because that's what I use myself — though I do it manually rather than in any kind of config file.

**Comment by AtkinsSJ** [comment]
This is very late, but if you run ladybird with the `--debug-process` flag, it'll launch, and then pause the named process until a debugger attaches. (eg, `--debug-process WebContent` is usually what you want)

I'm not sure if that's helpful here. I also don't know if the flag existed when you opened this issue.

**Comment by sideshowbarker** [comment]
The **Debugging** section of `Documentation/EditorConfiguration/VSCodeConfiguration.md` has this now — and also specifically mentions using `--debug-process WebContent` to attach to the right child process. Closing as completed.
